### ISSUE TITLE
Multible kernel issue due to the rolling repo

### ISSUE THREAD

**Issue by fruchttiger00x0** [g0tmi1k/os-scripts#6]
Hi together,

first of all, love your script, tasty stuff! But since the last couple of weeks it wont succeed anymore. If i start with a fresh amd64 image (full), it leads to a mess. I guess it's because of the the latest 4.2.0 image in the rolling repo. After step 'Updating OS' it will remove nearly the complete meta 'kali-linux-full' while its trying to upgrade the build. 
It will only succeed if I upgrade everything before running kali.sh or, like i did in my fork, if i comment out the source.list adaptation.  Could you please have a look on it? Many thx

p.s. sorry for my bad english :)

Edit:

Two things in addition. As a result of that issue the XFCE environment get some, lets say cosmetic defects. The first bug is that the Icons in the panel get lost but only that from the kali menue (like 'Information Gathering', 'Wireless Attacks' and so on). Tried to fix it with your XFCE Script from the board, g0tmi1k but it wont bring back the icons
"curl -s http://pastebin.com/raw.php?i=58wsGhBD | tr -d '\15\32' | bash"

Second one is the Conky Desktop Widget. There the transparency get lost and the background color went to green or red. This one i could fix in some way. I adjust these two settings:

- background yes
+ background no

- own_window_argb_visual yes   # GNOME & XFCE yes, KDE no
- own_window_argb_visual no   # GNOME & XFCE yes, KDE no

After all thats nothing too critically, sure but maybe the solution is relatively easy to adapt.

If i should deliver more details, please let me know

Kind regards

**Comment by g0tmi1k** [comment]
I also found out about the kernel & rolling issue the hard way yesterday... Been looking into a fix =).

**Comment by fruchttiger00x0** [comment]
Awesome! Thx for all that work

By the way, maybe you should add some routine that checks if repo already exists. I always have to clean up the source files. And yes,  now i'm getting greedy xD

Sorry, just kidding. I can also be helpful in some way. Let me know if i can support you in some way :)

One possibility for repo check

repo=kali-rolling

if ! grep -q "$repo" /etc/apt/sources.list /etc/apt/sources.list.d/*; then
    echo "deb http://http.kali.org/kali kali-rolling main contrib non-free" >> "${file}"
fi

**Comment by g0tmi1k** [comment]
It already does check?

```
grep -q 'deb .* kali-rolling main contrib non-free' "${file}" 2>/dev/null || echo "deb http://http.kali.org/kali kali-rolling main contrib non-free" >> "${file}"
```
Grep (quietly) to see if the rolling repo is there. If its not (due to the exit code not being 0), then echo the value into the file.

Source: https://github.com/g0tmi1k/os-scripts/commit/e697b4ba5177788a2ffbb5c5dcb9eb3bc186875e#diff-145a9b33a8f8af1d0079a975b63197d1R201

**Comment by fruchttiger00x0** [comment]
I just checked again and it is only the sec update repo. 
 +grep -q 'deb-src .* kali-rolling/updates main contrib non-free"' "${file}" 2>/dev/null || echo "deb http://security.kali.org/kali-security kali-rolling/updates main contrib non-free" >> "${file}" 

Look at the end of the search pattern. Quotation marks?

**Comment by g0tmi1k** [comment]
Ah! Then that will be a bug in the code.
Will fix it in the next commit.

Nice catch ;)

**Comment by fruchttiger00x0** [comment]
Yepp, lucky hit
And now where the main work is done, i can go to sleep and leave the easy ones for you guys 
hehe ;)

thx & see ya

**Comment by Indrasblade** [comment]
Hi, your tutorials are incredible, first class! 

I must apologise if I'm posting on an inappropriate thread, I'm a noob when it comes to pentesting and I've run into issues trying to install your OS scripts. I've been attempting to brute force the main DVWA web page for a while and noone other than yourself (and this includes seasoned pentesters) seemed to know or would let on that it's an anti-CSRF forgery preventing the standard burpsuite hydra attacks.
I'm really only after the part of the code that will allow me to see the hidden User token fields.
When i install the Kali.sh it hangs after a while and I have to exit out. I'm sorry I don't have the information on what stage it stopped at, I've updated/upgraded my OS to restore any compromised OS modules.
Is there an alternative way to get the hidden fields and follow through with your tutorial or an easy way to download your specific script containing the functionality required to view the tokens without the bells and whistles?

Once again, sorry if It's a stupid question and thanks for the tutorials.

**Comment by Indrasblade** [comment]
Hi

Your tutorials are excellent!

There seems to be an issue with the main DVWA login page hack. On my DVWA main login page there doesn't appear to be any hidden user token. I cannot seem to hack it with Hydra so I just assumed it must be the anti-CSRF token you mentioned but it' not showing up in Brpuite or with cURL. Any ideas?

**Comment by Indrasblade** [comment]
Hi

Your tutorials are excellent!

There seems to be an issue with the main DVWA login page hack. On my DVWA main login page there doesn't appear to be any hidden user token. I cannot seem to hack it with Hydra so I just assumed it must be the anti-CSRF token you mentioned but it' not showing up in Burpsuite or with cURL. Any ideas? (I'm on Kali 2.0 Gnome 3 host with the DVWA site hosted on my Virtualbox Metasploitable 2 guest.

**Comment by g0tmi1k** [comment]
@Indrasblade 
Off topic. 
DVWA has nothing todo with this.

**Comment by g0tmi1k** [comment]
Okay. Latest commit should fix a few of these issues.

Will start to look into conky more now =).

**Comment by fruchttiger00x0** [comment]
Great news! As soon as my live build finished i'll give it a try. And don't worry, setting up the OS 
is one of my favorite things to do. And of course I do this on dedicated clients. So no trouble :)
I'll give feeback soon

**Comment by Indrasblade** [comment]
Hi g0tmi1k, I love your tutorials, they're amazing. I'm a bit nooby when it comes to this stuff. I'm on Kali 2.0 Gnome and have had trouble trying to install your OS scripts, they error out and hang. I'm sorry I can't tell you what stage they error out at (not at all helpful I know) I've since reupdated and reupgraded my OS modules to ensure they weren't compromised by a half-install.
I attempted to install your scripts because in following your tutorial about hacking the DVWA I saw that my cURL commands or Burpsuite captures weren't showing any anti-CSRF user tokens like the tutorial showed. I assumed it was because I needed your scripts to see them.

I've been having problems getting Hydra to work on the main DVWA login page with a failed login string and I assumed it's because of anti CSRF tokens that it wasn't working. I've since found I can get Hydra to work with a success string 'S=index' representing the redirect to the page in a successful login. I guess this means it wasn't CSRF tokens stopping me logging in as they would also prevent successful condition logins.
So is there a setting in DVWA to enable CSRF tokens to follow your tutorial? Or are the CSRF tokens actually present but I'm just not seeing them? Also do you know why so many people are having trouble with the main login page and a failed string condition?

**Comment by fruchttiger00x0** [comment]
Put this thing to kali board or somewhere it belongs. I could help you in a way but i want till make it right. He want help you cause its not a issue related to that script. This thing pulls Installation subroutines, do the dispatching and is adding framework, extension etc. Your question is not postinstallation kali, end of story.
@g0tm1lk
Please remove our postings

**Comment by g0tmi1k** [comment]
@Indrasblade 
Once again, off topic.
As its about DVWA and NOT about this repo - ask your question in the RIGHT place: https://github.com/RandomStorm/DVWA

**Comment by fruchttiger00x0** [comment]
I was able to set up a pxe setup to install base system + curl and a trigger to append kali.sh before i went to bed. So this morning I had 5 minutes for smoke testing but it seems that you have done a good job. Script was running 293 minutes on my HP 840 Elitebook (core i5, SSD, 8GB RAM) and a 16Mbit Downlink. So without looking to deep I would guess : works as designed 
Many THX!

**Comment by g0tmi1k** [comment]
It mainly depends on your Internet connection & location.
+ Without rolling takes me about ~70 minutes 
+ With rolling takes me about ~130 minutes

**Comment by Indrasblade** [comment]
Ok, thanks for the help guys.

**Comment by Indrasblade** [comment]
Ok, thanks for the help guys. Sorry about the mispost.

**Comment by fruchttiger00x0** [comment]
Hi,
after testing my PXE Installation a while I found some over minor bugs (for example conky and the icons again). But as they all adress kali-rolling or better say the kernel 4.2.0 I would agree if we close the issue.  Script is working well cause Rolling becomes optinal. Again, very good job.

So I did take another run yesterday. This time with a basic installation which tooks 83 min:
kali-linux-2.0-amd64 followed by kali.sh without --rolling

No complaints this time. Issue is solved for me.

**Comment by bb1ox1bb** [comment]
i dont know how much that is usefull for u guys but have a look at that 

[+] Installing wifite ~ automated Wi-Fi tool
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get

 [+] Installing vulscan script for nmap ~ vulnerability scanner add-on
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get
######################################################################## 100.0%

 [+] Installing unicornscan ~ fast port scanner
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get

 [+] Installing onetwopunch ~ unicornscan & nmap wrapper
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get

 [+] Installing Gnmap-Parser (Fork) ~ Parse Nmap exports into various plain-text formats
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get
chmod: cannot access ‘/opt/gnmap-parser-git/Gnmap-Parser.sh’: No such file or directory

 [+] Installing udp-proto-scanner ~ common UDP port scanner
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get
######################################################################## 100.0%

 [+] Installing clusterd ~ clustered attack toolkit (JBoss, ColdFusion, WebLogic, Tomcat etc)
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get

 [+] Installing webhandler ~ shell TTY handler
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get

 [+] Installing azazel ~ Linux userland rootkit
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get

 [+] Installing Babadook ~ connection-less powershell backdoor
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get

 [+] Installing pupy ~ Remote Administration Tool
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
 [!] Issue with apt-get

and all of them going like that ...

deb http://http.kali.org/kali sana main non-free contrib
deb http://security.kali.org/kali-security sana/updates main contrib non-free

deb http://http.kali.org/kali kali-rolling main contrib non-free
deb http://security.kali.org/kali-security kali-rolling/updates main contrib non-free
deb-src http://http.kali.org/kali sana main non-free contrib
deb-src http://security.kali.org/kali-security sana/updates main contrib non-free

any solution? or where im going wrong?

**Comment by g0tmi1k** [comment]
@bb1ox1bb
New issue, new thread.
Please guys try and keep on topic....

**Comment by bb1ox1bb** [comment]
okay then finish install and open new issuse :+1:

**Comment by freis** [comment]
@fruchttiger00x0 you got any idea how to get around that bugs? the icons and such?

**Comment by fruchttiger00x0** [comment]
Hi, as g0tm1lk told the problem comes with incompatibilities between kernel 4.2. and Windows manager, the debugging should not be our part. So which workarround you take is your decision. If you love xfce and not realy need the latest package  version from the rolling repository then just run the script without --rolling Option. Otherwise switch to GNOME after script finished to workarround that bugs.

**Comment by freis** [comment]
@fruchttiger00x0 well i guess i'll go without rolling option. really dont like kali with all that stuff, thanks

**Comment by fruchttiger00x0** [comment]
I installed an instance of kali on several devices cause for some reason I like doing that, xD But nevermind the point is that you almost need to be flexible and prepared when you want to do more then playing arround with one or two tools in kali. The rolling repo is for having the latest version of a tool. And beeing a step ahead is is essential if you just think of exploits for example. In a nutshell: kali is a very handsome Linux for personal issues and state of the Art for special treatment. Why not having at least two installations :) hope you got it

**Comment by g0tmi1k** [comment]
@freis  Regarding kali-rolling:

```
Note however that those checks do not include any functional testing. It might still contain broken software due to other problems that are not covered by the package dependencies. Advanced Kali users that are confident in their abilities to work around/recover from such problems are free to use Kali Rolling to benefit from the latest and greatest software that Debian and Kali provides.
```

Source: http://docs.kali.org/general-use/kali-linux-sources-list-repositories

In short, **expect bugs - meant for advanced linux users**.

**Comment by famous0123** [comment]
please sir i have Error "Issue With apt-get " what can i do ?

**Comment by DOOMexe** [comment]
I just trying running this on a clean install on Kali Rolling.
I did a `apt-get update`, `apt-get upgrade and `apt-get dist-upgrade` and then rebooted.

When I ran `./kali.sh -burp -openvas -rolling -keyboard us -timezone "Europe/Berlin"`, I seeing the below errors.


```
 [+] Installing kernel headers
E: Unable to locate package linux-headers-4.3.0-kali1-amd64
E: Couldn't find any package by glob 'linux-headers-4.3.0-kali1-amd64'
E: Couldn't find any package by regex 'linux-headers-4.3.0-kali1-amd64'
 [!] Issue with apt-get
```

But then see the below a few lines down.

` [+] Enabling rolling repositories ~ Should only be used by advanced users! Using this means tools will be updated more frequently`
